audinterface/core/feature.py

```diff
@@ -567,6 +579,15 @@ def process_index(
             if cache_path is not None:
                 df.to_pickle(cache_path, protocol=4)
 
+        if self.process.segment is None and preserve_index:
+            # Convert segmented index to filewise index
+            # if original index was filewise
+            if index_type == audformat.define.IndexType.FILEWISE:
```

Comment:
Can't we simply do: `df.index = index`?

Reply:
You are right.
I updated the code here and in `process.py`.

Reply:
Thanks. Since you are no longer checking if the input is a filewise index, this will also preserve `NaT` values for segmented indices, which I think is what a user would expect. But as far as I see this case is not covered by the tests, yet.

Reply:
I added the `NaT` to the tests and restructured them to provide `preserve_index` as a pytest parameter.